### PR TITLE
Oracle for LP

### PR DESCRIPTION
**Pull request by shouc**

### CODE REVIEW

File: src/evm/oracles/erc20.rs [hunk 1]
```diff
@@ -249,8 +192,8 @@ impl Oracle<EVMState, H160, Bytecode, Bytes, H160, U256, Vec<u8>, EVMInput, EVMF
         {
             let net = exec_res.new_state.state.flashloan_data.earned
                 - exec_res.new_state.state.flashloan_data.owed;
-            // we scaled by 1e23, so divide by 1e23 to get ETH
-            let net_eth = net / U512::from(1_000_000_000_000_000_000_000_00u128);
+            // we scaled by 1e24, so divide by 1e24 to get ETH
```
Comment (stanfordreject) [inline src/evm/oracles/erc20.rs:195]: ?

File: src/evm/producers/pair.rs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                .fuzz_state
+                .get_execution_result()
+                .new_state
+                .state
+                .flashloan_data
+                .oracle_recheck_reserve
+                .clone();
+            let mut query_reserves_batch = reserves.iter().map(
+                |pair_address| {
+                    (*pair_address, Bytes::from(vec![0x09, 0x02, 0xf1, 0xac]))
```
Comment (stanfordreject) [inline src/evm/producers/pair.rs:52]: move bytes from out of the loop?

File: src/evm/producers/erc20.rs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            let tokens = ctx
+                .fuzz_state
+                .get_execution_result()
+                .new_state
+                .state
+                .flashloan_data
+                .oracle_recheck_balance
+                .clone();
+
+            let callers = ctx.fuzz_state.callers_pool.clone();
```
Comment (stanfordreject) [inline src/evm/producers/erc20.rs:54]: move it into self?

File: src/evm/oracles/v2_pair.rs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        #[cfg(feature = "flashloan_v2")]
+        {
+            let prev_reserves = ctx
+                .fuzz_state
+                .get_execution_result()
+                .new_state
+                .state
+                .flashloan_data
+                .prev_reserves
+                .clone();
```
Comment (stanfordreject) [inline src/evm/oracles/v2_pair.rs:59]: clone not needed

File: src/evm/oracles/erc20.rs (hunk 1 above)
Comment (shouc) [reply src/evm/oracles/erc20.rs:195]: 1e6 x 1e18

File: src/evm/producers/erc20.rs (hunk 3 above)
Comment (shouc) [reply src/evm/producers/erc20.rs:54]: callers_pool is dynamic

File: src/evm/producers/pair.rs (hunk 2 above)
Comment (shouc) [reply src/evm/producers/pair.rs]: fixed

File: src/evm/oracles/v2_pair.rs (hunk 4 above)
Comment (shouc) [reply src/evm/oracles/v2_pair.rs]: fixed

**Comment by stanfordreject** [comment]
lgtm

**Pull request merged**